src/services/ImportContactsService.ts

```diff
@@ -27,17 +22,17 @@ class ImportContactsService {
     const createdTags = await Tag.create(newTagsData);
     const tagsIds = createdTags.map(tag => tag._id);
 
-    parseCSV.on('data', async line => {
-      const [email] = line;
-
-      await Contact.findOneAndUpdate(
-        { email },
-        { $addToSet: { tags: tagsIds } },
-        { upsert: true },
-      );
-    });
-
-    await new Promise(resolve => parseCSV.on('end', resolve));
+    const emails = await parseCSVAsync(contactsFileStream);
```

Comment:
This way, all the contacts will only be registered after the CSV was read by full, so we are not using the power of the streams to process the data while reading it. What do you think about it?

Reply:
I can't believe I forgot it, the files can be very large, to resolve, maybe a function param will called each a new line is readed, but keeping the resolve at the function end to only proceed if it's end?

Reply:
I think if we keep the resolve at the end, it will be called right after the function call, but we need to know when all the process ends. We could separate the `parseCSVAsync` logic as you doing but we need to process each line of the CSV with the stream and also knows when it ends so we can return a response :)

Reply:
Oh sorry, I wrote wrong on my last comment, the resolve only be called on end listener of parseCsv's parser. I did a test to see if the function parameter works and stayed like this:

`ImportContactsService.ts`
```js
// ...

    await parseCSVAsync(contactsFileStream, () => console.log('line'));

     console.log('end');

// ...
```

`parseCSVAsync.ts`
```js
import { Readable } from 'stream';
import csvParse from 'csv-parse';

export default function parseCSVAsync(
  fileStream: Readable,
  eachLine?: (...line: string[]) => Promise<void> | void, // `...line` or only `line`?
): Promise<string[]> {
  return new Promise((resolve, reject) => {
    const result: string[] = [];

    const parsers = csvParse({
      delimiter: ';',
    });

    const parseCSV = fileStream.pipe(parsers);

    parseCSV.on('data', async (line: string[]) => {
      result.push(...line);

      await eachLine?.(...line);
    });

    parseCSV.on('end', () => resolve(result));

    parseCSV.on('error', reject);
  });
}
```

And it worked well, results:
![image](https://user-images.githubusercontent.com/29582786/72994014-1ce01f80-3dd5-11ea-99ae-350a3e071a0e.png)

Reply:
That's great! Can you merge this inside this PR? You can amend to the last commit and `git push` with force (`-f`) option, so we won't have two commits :)

Reply:
Merged!